metadata-ingestion/src/datahub/ingestion/graph/client.py

```diff
@@ -738,32 +751,174 @@ def get_urns_by_filter(
             """
         )
 
+        variables = {
+            "types": types,
+            "query": query,
+            "orFilters": orFilters,
+            "batchSize": batch_size,
+        }
+
+        for entity in self._scroll_across_entities(graphql_query, variables):
+            yield entity["urn"]
+
+    def _scroll_across_entities(
+        self, graphql_query: str, variables: dict
+    ) -> Iterable[dict]:
         first_iter = True
         scroll_id: Optional[str] = None
         while first_iter or scroll_id:
             first_iter = False
+            variables["scrollId"] = scroll_id
```

Comment:
copy variables to avoid mutating the dict that the use passed in

Reply:
done